Tasks/Gradle/CodeAnalysis/Common/FileSystemInteractions.ts

```diff
@@ -0,0 +1,86 @@
+import shell = require('shelljs');
+import path = require('path');
+import fs = require('fs');
+
+import tl = require('vsts-task-lib/task');
+
+export class FileSystemInteractions {
+    public static copyFile(sourcePath:string, destinationPath:string):void {
+        shell.cp('-f', sourcePath, destinationPath);
+
+        this.checkShell('cp', false);
+    }
+
+    /**
+     * Create a directory at the specified path, including any folders in between.
+     * @param directoryPath Path to create.
+     */
+    public static createDirectory(directoryPath:string):void {
```

Comment:
Are there any unit tests that can be copied for this? There is quite a lot of code here. If not, can you add some basic unit tests?